contracts/events/TopicEvent.sol

```diff
@@ -299,54 +292,46 @@ contract TopicEvent is ITopicEvent, BaseContract, Ownable {
     }
 
     /* 
-    * @notice Calculates the QTUM tokens won based on the sender's QTUM token contributions.
-    * @return The amount of QTUM tokens won.
+    * @notice Calculates the BOT and QTUM tokens won based on the sender's contributions.
+    * @return The amount of BOT and QTUM tokens won.
     */
-    function calculateQtumContributorWinnings() 
+    function calculateWinnings()
         public 
         view
         inCollectionStatus()
-        returns (uint256)  
+        returns (uint256, uint256)  
     {
-        uint256 senderContribution = balances[resultIndex].bets[msg.sender];
+        uint256 votes = balances[resultIndex].votes[msg.sender];
+        uint256 bets = balances[resultIndex].bets[msg.sender];
+        if (votes == 0 && bets == 0) {
```

Comment:
This check doesn't skip bet or vote calculate when one of "bets" or "votes" is zero. My suggestion:

rewardQtum = 0;

if(bets != 0 )
{
// calc bets
}

if(votes !=0){

// calc votes
}

Reply:
ah i got you. Fixed.